app/src/ui/preferences/confirm-restart.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react'
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  OkCancelButtonGroup,
+} from '../dialog'
+import { restartApp } from '../main-process-proxy'
+
+interface IConfirmRestartProps {
+  /**
+   * Callback to use when the dialog gets closed.
+   */
+  readonly onDismissed: () => void
+}
+
+export class ConfirmRestart extends React.Component<IConfirmRestartProps> {
+  public constructor(props: IConfirmRestartProps) {
+    super(props)
+  }
+
+  public render() {
+    return (
+      <Dialog
+        dismissable={false}
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.onSubmit}
+        type="warning"
+      >
+        <DialogContent>
+          <p>Restart GitHub Desktop to apply the change. </p>
+          <div></div>
```

Comment:
:art: let's polish this message a bit

```suggestion
          <p>Restart GitHub Desktop to apply the title bar settings change?</p>
```

Screenshot:

![Screenshot from 2023-08-06 10-28-05](https://github.com/shiftkey/desktop/assets/359239/2bf90050-dd3a-42ce-9122-f253f91d3f3f)